src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -44,9 +44,10 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             {
```

Comment:
The condition at line 43 should be probably more like: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/161411bf059185d8b4e20d7cae0d6c3d29b2da13/src/client/Microsoft.Identity.Client/Internal/Requests/Silent/CacheSilentStrategy.cs#L46